dali/operators/decoder/audio/audio_decoder_op.cc

```diff
@@ -98,74 +91,40 @@ AudioDecoderCpu::SetupImpl(std::vector<OutputDesc> &output_desc, const workspace
+  auto &meta = sample_meta_[sample_idx];
+  float target_sr = use_resampling_ ? target_sample_rates_[sample_idx] : meta.sample_rate;
+  bool should_resample = target_sr != meta.sample_rate;
   bool should_downmix = meta.channels > 1 && downmix_;
-  if (should_resample || should_downmix || output_type_ != decode_type_) {
-    assert(decode_type_ == DALI_FLOAT);
-    int64_t tmp_size = should_downmix && should_resample
-      ? meta.length * (meta.channels + 1)   // downmix to intermediate buffer, then resample
-      : meta.length * meta.channels;        // decode to intermediate, then resample or downmix
-                                            // directly to the output
-
-    tmp_buf.resize(tmp_size);
-    size_t num_samples = meta.length * meta.channels;
-    size_t ret = decoders_[sample_idx]->Decode(as_raw_span(tmp_buf.data(), num_samples));
-    DALI_ENFORCE(ret == num_samples,
-                 make_string("Error decoding audio file: ", files_names_[sample_idx]));
-
-    if (should_downmix) {
-      if (should_resample) {
-        // downmix and resample
-        float *downmixed = tmp_buf.data() + meta.length * meta.channels;
-        assert(downmixed + meta.length <= tmp_buf.data() + tmp_buf.size());
-        kernels::signal::Downmix(downmixed, tmp_buf.data(), meta.length, meta.channels);
-        resampler_.Resample(audio.data, 0, audio.shape[0], output_rate,
-                            downmixed, meta.length, meta.sample_rate);
-      } else {
-        // downmix only
-        kernels::signal::Downmix(audio.data, tmp_buf.data(), meta.length, meta.channels);
-      }
-    } else if (should_resample) {
-      // multi-channel resample
-      resampler_.Resample(audio.data, 0, audio.shape[0], output_rate,
-                          tmp_buf.data(), meta.length, meta.sample_rate, meta.channels);
-
-    } else {
-      // convert or copy only - this will only happen if resampling is specified, but this
-      // recording's sampling rate and number of channels coincides with the target
-      int64_t len = std::min<int64_t>(volume(audio.shape), meta.length*meta.channels);
-      for (int64_t ofs = 0; ofs < len; ofs++) {
-        audio.data[ofs] = ConvertSatNorm<OutputType>(tmp_buf[ofs]);
-      }
-    }
-  } else {
-    assert(!should_downmix && !should_resample);
-    size_t num_samples = volume(audio.shape);
-    size_t ret = decoders_[sample_idx]->Decode(as_raw_span(audio.data, num_samples));
-    DALI_ENFORCE(ret == num_samples,
-                 make_string("Error decoding audio file: ", files_names_[sample_idx]));
-  }
+  int64_t decode_scratch_sz = 0;
+  int64_t resample_scratch_sz = 0;
+  if (should_resample || should_downmix || !std::is_same<OutputType, DecoderType>::value)
+    decode_scratch_sz = meta.length * meta.channels;
+
+  // resample scratch is used to prepare a single or multiple (depending if
+  // downmixing is needed) channel float input, required by the resampling
+  // kernel
```

Comment:
```suggestion
  /*
    * resample scratch is used to prepare a single or multiple (depending if
    * downmixing is needed) channel float input, required by the resampling
    * kernel
    */
```

Reply:
why? it is a cc file

Reply:
Haven't we agree to just use this for multiline comments, no matter if cc or h?

Reply:
I don't recall such thing. I think it's just for headers. Anyway, if you insist I can change it

Reply:
It was only for doxygen.

Reply:
> Haven't we agree t
/ rant / ...we didn't _agree_. I might have _adjusted_ , but I've never _agreed_, much less undersetood this particular choice.